### PR TITLE
added fastapi.yml and updated django and flask

### PR DESCRIPTION
**Pull request by Genegrady**
## Checklist
- [ ] Added to the correct milestone.
- [ ] Tests provided or description of manual testing performed is included in the code or PR.
- [ ] Library documentation is updated.
- [ ] [Corp site](https://github.com/DataDog/documentation/) documentation is updated (link to the PR).

### CODE REVIEW

File: .github/workflows/django.yml [hunk 1]
```diff
@@ -20,7 +20,7 @@ jobs:
           path: django
       - uses: actions/setup-python@v2
         with:
-          python-version: "3.8"
+          python-version: '3.8'
```
Comment (brettlangdon) [inline .github/workflows/django.yml:23]: can you revert this change or make this in another PR?

File: .github/workflows/fastapi.yml [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+          path: fastapi
+      - name: Install Flit
+        run: pip install flit
+      - name: Inject ddtrace
+        run: pip install ddtrace
+      - name: Install Dependencies
+        run: flit install --symlink
+      - name: Test
+        run: bash scripts/test.sh
+      #- name: Upload coverage
```
Comment (brettlangdon) [inline .github/workflows/fastapi.yml:39]: we won't need this

File: .github/workflows/fastapi.yml [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+  pull_request:
+jobs:
+  fastapi-test:
+    runs-on: ubuntu-latest
+    defaults:
+      run:
+        working-directory: fastapi
+    strategy:
+      matrix:
+        python-version: [3.6, 3.7, 3.8]
```
Comment (Kyle-Verhoog) [inline .github/workflows/fastapi.yml:15]:
```suggestion
        python-version: [3.6, 3.7, 3.8, 3.9]
```

File: .github/workflows/fastapi.yml [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+          ref: 0.63.0
+          path: fastapi
+      - name: Install Flit
+        run: pip install flit
+      - name: Inject ddtrace
+        run: pip install ddtrace
+      - name: Install Dependencies
+        run: flit install --symlink
+      - name: Test
+        run: bash scripts/test.sh
```
Comment (Kyle-Verhoog) [inline .github/workflows/fastapi.yml:38]: We probably need to `ddtrace-run` this script in order to actually inject the tracer (as it stands it doesn't look like the tracer will be used in the test suite)

File: .github/workflows/fastapi.yml (hunk 2 above)
Comment (Yun-Kim) [reply .github/workflows/fastapi.yml:39]: Where do we use ddtrace to run the tests?

File: .github/workflows/django.yml (hunk 1 above)
Comment (Genegrady) [reply .github/workflows/django.yml:23]: Will do!

File: .github/workflows/fastapi.yml [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+        with:
+          python-version: ${{ matrix.python-version }}
+      - uses: actions/checkout@v2
+        with:
+          repository: tiangelo/fastapi
+          ref: 0.63.0
+          path: fastapi
+      - name: Install Flit
+        run: pip install flit
+      - name: Inject ddtrace
```
Comment (majorgreys) [inline .github/workflows/fastapi.yml:33]: We want to do something similar to what's done in the other workflows for checking out the repo and installing from the path since we want to run this workflow for a commit rather than what is currently deployed to pypi.

File: .github/workflows/fastapi.yml [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+      - name: Install Dependencies
+        run: flit install --symlink
+      - name: update mypy
+        run: pip install --upgrade --force-reinstall mypy
+      - name: Install Typing
+        run: pip install typing
+      - name: Inject ddtrace
+        run: pip install ddtrace
+      - name: Test
+        run: ddtrace-run scripts/test.sh
```
Comment (brettlangdon) [inline .github/workflows/fastapi.yml:41]: we might want to use something other than `test.sh` here, not sure we want to run their linting?

Comment (brettlangdon) [reply .github/workflows/fastapi.yml:41]: e.g. instead of calling `scripts/test.sh` we just manually call these two lines:

https://github.com/tiangolo/fastapi/blob/22528373bba6a654323de416ad5c867cbadb81bb/scripts/test.sh#L9-L10

we can even modify to remove code coverage reporting, so it will really just be `pytest tests`

and if we control that command (rather than calling through `scripts/test.sh` then we can add our ddtrace plugin to pytest to enable patching/tracing.

File: .github/workflows/fastapi.yml (hunk 3 above)
Comment (Yun-Kim) [reply .github/workflows/fastapi.yml]: Just curious, @Kyle-Verhoog @Genegrady is it worth testing the fastapi tests on 4 different python versions?

File: .github/workflows/fastapi.yml [hunk 7]
```diff
@@ -0,0 +1,42 @@
...
+      - name: Install Flit
+        run: pip install flit
+      - name: Install Dependencies
+        run: flit install --symlink
+      - name: Downgrade SQLalchemy
+        run: pip install --force-reinstall SQLAlchemy==1.3.24
+      - name: Install Typing
+        run: pip install typing
+      - name: Inject ddtrace
+        run: pip install ddtrace
```
Comment (Yun-Kim) [inline .github/workflows/fastapi.yml:39]:
```suggestion
        run: pip install ../ddtrace
```
We typically want to run the tests on the modified code being submitted in the PR instead of pulling from pypi 😄

Comment (Genegrady) [reply .github/workflows/fastapi.yml]: Just updated to this code, now Python version 3.7 and 3.8 are breaking when installing dd-trace

Comment (Genegrady) [reply .github/workflows/fastapi.yml]: @Yun-Kim

Comment (Yun-Kim) [reply .github/workflows/fastapi.yml]: Seems like this is a `typing`-related issue that stems from `typing` being in the standard python library starting from 3.5, can see more here: https://github.com/python/typing/issues/573
What was the reason we added the install typing step?

Comment (Genegrady) [reply .github/workflows/fastapi.yml]: @Yun-Kim Originally installed bc tests were failing locally without it. Will remove to see if this causes tests to succeed.

Comment (Genegrady) [reply .github/workflows/fastapi.yml]: @Yun-Kim Tests are now succeeding for fastapi, but failing other tests (flask, circleci)

Comment (brettlangdon) [reply .github/workflows/fastapi.yml]: those two CircleCI ones are a known issue we are working through right now, see #2371

for the flask framework test suite, you can ignore for now, it has been a little flaky. something we need to resolve (that specific failure will not block this PR... CircleCI ones will though)

Comment (Genegrady) [reply .github/workflows/fastapi.yml]: @brettlangdon thanks for the clarification! I will work on other tests while I wait for the Circleci issues to resolve.

Comment (brettlangdon) [reply .github/workflows/fastapi.yml]: should be resolved now if you update from `master`

File: .github/workflows/fastapi.yml [hunk 8]
```diff
@@ -0,0 +1,40 @@
...
+  pull_request:
+jobs:
+  fastapi-test:
+    runs-on: ubuntu-latest
+    defaults:
+      run:
+        working-directory: fastapi
+    strategy:
+      matrix:
+        python-version: [3.6, 3.7, 3.8, 3.9]
```
Comment (brettlangdon) [inline .github/workflows/fastapi.yml:15]: think I'll echo a previous comment, we can probably just test 1 version of Python, since FastAPI is testing that they are compatible with all of these versions, we just need to be sure we are compatible with FastAPI

(unless our instrumentation has a bunch of python version conditionals?)

Comment (Genegrady) [reply .github/workflows/fastapi.yml:15]: @brettlangdon Sounds good. I will change the conditional to the latest version (3.9)

File: .github/workflows/fastapi.yml [hunk 9]
```diff
@@ -0,0 +1,40 @@
...
+        working-directory: fastapi
+    strategy:
+      matrix:
+        python-version: [3.9]
+      fail-fast: false
+    steps:
+      - name: Set up Python
+        uses: actions/setup-python@v2
+        with:
+          python-version: ${{ matrix.python-version }}
```
Comment (Yun-Kim) [inline .github/workflows/fastapi.yml:21]:
```suggestion
    steps:
      - uses: actions/setup-python@v2
        with:
          python-version: '3.9'
```
This way, we can also get rid of the `strategy: ...` block as well and make this simpler 😄

Comment (Genegrady) [reply .github/workflows/fastapi.yml]: Updated 😄

File: .github/workflows/fastapi.yml [hunk 10]
```diff
@@ -0,0 +1,35 @@
...
+        with:
+          repository: tiangolo/fastapi
+          ref: 0.63.0
+          path: fastapi
+      - name: Install Flit
+        run: pip install flit
+      - name: Install Dependencies
+        run: flit install --symlink
+      - name: Downgrade SQLalchemy
+        run: pip install --force-reinstall SQLAlchemy==1.3.24
```
Comment (Yun-Kim) [inline .github/workflows/fastapi.yml:30]: It would be nice if you could add a quick comment above each of these these steps to label as being fastapi specific, and maybe explain why these steps are necessary to provide some context in the future 😄

**Review by Yun-Kim — COMMENTED** [review]
Looks good to me once some comments are added to the workflow. Just curious, what does the output of the tests look like when you run this action locally?

**Review by Genegrady — COMMENTED** [review]
Added comments to FastAPI specific steps.

**Review by Kyle-Verhoog — APPROVED** [review]
👏 lgtm

**Review by brettlangdon — APPROVED** [review]
